src/renderers/WebGLRenderer.js

```diff
@@ -1339,13 +1339,27 @@ class WebGLRenderer {
 
 			const isWebGL2 = capabilities.isWebGL2;
 
+			const currentRenderTarget = _this.getRenderTarget();
+
+			// If the current render target has a stencil buffer, then the transmission render target also needs it so that it's rendered the same way.
+
+			const needStencilBuffer = currentRenderTarget !== null ? currentRenderTarget.stencilBuffer : _this.getContextAttributes().stencil;
+
+			if ( _transmissionRenderTarget !== null && needStencilBuffer && ! _transmissionRenderTarget.stencilBuffer ) {
+
+				_transmissionRenderTarget.dispose();
+				_transmissionRenderTarget = null;
+
+			}
+
 			if ( _transmissionRenderTarget === null ) {
 
 				_transmissionRenderTarget = new WebGLRenderTarget( 1, 1, {
 					generateMipmaps: true,
 					type: extensions.has( 'EXT_color_buffer_half_float' ) ? HalfFloatType : UnsignedByteType,
 					minFilter: LinearMipmapLinearFilter,
-					samples: ( isWebGL2 ) ? 4 : 0
+					samples: ( isWebGL2 ) ? 4 : 0,
+					stencilBuffer: needStencilBuffer
```

Comment:
For simplicity reasons, couldn't we just always create the transmission render target with a stencil buffer? I don't think this would cause noticeable overhead.

Reply:
I'd always avoid extra memory allocations wherever possible. On a large resolution screen the bytes add up.

Reply:
TBH, I'm still a bit unsure about the amount of new code and the checks per frame if we could also fix the issue with just `stencilBuffer: true`.

is it possible to quantify how much memory the engine would additionally allocate with enabled stencil buffer and e..g a 5K framebuffer ?

Reply:
Stencil uses 1 byte per pixel, so for rendering at 5K resolution that's about 15 megabytes. Whether that's a lot or not depends on perspective, but I think we should avoid this kind of waste. With WebGL it also can't be assumed that the user is focusing on one app at once, but rather it's quite common to have multiple open simultaneously.

Reply:
> With WebGL it also can't be assumed that the user is focusing on one app at once, but rather it's quite common to have multiple open simultaneously.

That's true but only when all applications actually rendering transmissive objects (which is not a common thing I would say). But yes I understand the concerns regarding memory.

@mrdoob What is your preference regarding this issue?